File: sdk/python/test/e2e/rest/futures_test/futures_orders_test.py
import os
import unittest
import uuid

from kucoin_universal_sdk.api.client import DefaultClient
from kucoin_universal_sdk.extension.interceptor.logging import LoggingInterceptor
from kucoin_universal_sdk.generate.futures.order.model_add_order_req import AddOrderReqBuilder, AddOrderReq
from kucoin_universal_sdk.generate.futures.order.model_add_order_test_req import AddOrderTestReqBuilder, AddOrderTestReq
from kucoin_universal_sdk.generate.futures.order.model_add_tpsl_order_req import AddTpslOrderReqBuilder, AddTpslOrderReq
from kucoin_universal_sdk.generate.futures.order.model_batch_add_orders_item import BatchAddOrdersItemBuilder, \
    BatchAddOrdersItem
from kucoin_universal_sdk.generate.futures.order.model_batch_add_orders_req import BatchAddOrdersReqBuilder
from kucoin_universal_sdk.generate.futures.order.model_batch_cancel_orders_client_oids_list import \
    BatchCancelOrdersClientOidsListBuilder
from kucoin_universal_sdk.generate.futures.order.model_batch_cancel_orders_req import BatchCancelOrdersReqBuilder
from kucoin_universal_sdk.generate.futures.order.model_cancel_all_orders_req import CancelAllOrdersReqBuilder
from kucoin_universal_sdk.generate.futures.order.model_cancel_all_stop_orders_req import CancelAllStopOrdersReqBuilder
from kucoin_universal_sdk.generate.futures.order.model_cancel_order_by_client_oid_req import \
    CancelOrderByClientOidReqBuilder
from kucoin_universal_sdk.generate.futures.order.model_cancel_order_by_id_req import CancelOrderByIdReqBuilder
from kucoin_universal_sdk.generate.futures.order.model_get_open_order_value_req import GetOpenOrderValueReqBuilder
from kucoin_universal_sdk.generate.futures.order.model_get_order_by_client_oid_req import GetOrderByClientOidReqBuilder
from kucoin_universal_sdk.generate.futures.order.model_get_order_by_order_id_req import GetOrderByOrderIdReqBuilder
from kucoin_universal_sdk.generate.futures.order.model_get_order_list_req import GetOrderListReqBuilder, GetOrderListReq
from kucoin_universal_sdk.generate.futures.order.model_get_recent_closed_orders_req import \
    GetRecentClosedOrdersReqBuilder
from kucoin_universal_sdk.generate.futures.order.model_get_recent_trade_history_req import \
    GetRecentTradeHistoryReqBuilder
from kucoin_universal_sdk.generate.futures.order.model_get_stop_order_list_req import GetStopOrderListReqBuilder, \
    GetStopOrderListReq
from kucoin_universal_sdk.generate.futures.order.model_get_trade_history_req import GetTradeHistoryReqBuilder, \
    GetTradeHistoryReq
from kucoin_universal_sdk.model.client_option import ClientOptionBuilder
from kucoin_universal_sdk.model.constants import GLOBAL_API_ENDPOINT, GLOBAL_BROKER_API_ENDPOINT, \
    GLOBAL_FUTURES_API_ENDPOINT
from kucoin_universal_sdk.model.transport_option import TransportOptionBuilder


class FuturesApiTest(unittest.TestCase):
    def setUp(self):
        key = os.getenv("API_KEY")
        secret = os.getenv("API_SECRET")
        passphrase = os.getenv("API_PASSPHRASE")

        # create http transport option
        http_transport_option = (
            TransportOptionBuilder()
            .set_interceptors([LoggingInterceptor()])
            .build()
        )

        # create client option
        client_option = (
            ClientOptionBuilder()
            .set_key(key)
            .set_secret(secret)
            .set_passphrase(passphrase)
            .set_spot_endpoint(GLOBAL_API_ENDPOINT)
            .set_futures_endpoint(GLOBAL_FUTURES_API_ENDPOINT)
            .set_broker_endpoint(GLOBAL_BROKER_API_ENDPOINT)
            .set_transport_option(http_transport_option)
            .build()
        )

        # create API client
        client = DefaultClient(client_option)
        kucoin_rest_api = client.rest_service()
        self.api = kucoin_rest_api.get_futures_service().get_order_api()

    def test_get_trade_history_req(self):
        """
            get_trade_history
            Get Trade History
            /api/v1/fills
        """

        builder = GetTradeHistoryReqBuilder()
        builder.set_side(GetTradeHistoryReq.SideEnum.BUY).set_type(GetTradeHistoryReq.TypeEnum.MARKET).set_start_at(
            1732550400000).set_end_at(1732723200000)
        req = builder.build()
        try:
            resp = self.api.get_trade_history(req)
            print("code: ", resp.common_response.code)
            print("msg: ", resp.common_response.message)
            print("data: ", resp.to_json())
        except Exception as e:
            print("error: ", e)
            raise e

    def test_get_open_order_value_req(self):
        """
            get_open_order_value
            Get Open Order Value
            /api/v1/openOrderStatistics
        """

        builder = GetOpenOrderValueReqBuilder()
        builder.set_symbol("DOGEUSDTM")
        req = builder.build()
        try:
            resp = self.api.get_open_order_value(req)
            print("code: ", resp.common_response.code)
            print("msg: ", resp.common_response.message)
            print("data: ", resp.to_json())
        except Exception as e:
            print("error: ", e)
            raise e

    def test_get_order_by_client_oid_req(self):
        """
            get_order_by_client_oid
            Get Order By ClientOid
            /api/v1/orders/byClientOid
        """

        builder = GetOrderByClientOidReqBuilder()
        builder.set_client_oid("df2e52bb-1d11-44d9-91a9-26899f3f46cb")
        req = builder.build()
        try:
            resp = self.api.get_order_by_client_oid(req)
            print("code: ", resp.common_response.code)
            print("msg: ", resp.common_response.message)
            print("data: ", resp.to_json())
        except Exception as e:
            print("error: ", e)
            raise e

    def test_cancel_order_by_client_oid_req(self):
        """
            cancel_order_by_client_oid
            Cancel Order By ClientOid
            /api/v1/orders/client-order/{clientOid}
        """

        builder = CancelOrderByClientOidReqBuilder()
        builder.set_symbol("XBTUSDTM").set_client_oid("340cbc2a-13a7-468a-85e7-d05a9c607a67")
        req = builder.build()
        try:
            resp = self.api.cancel_order_by_client_oid(req)
            print("code: ", resp.common_response.code)
            print("msg: ", resp.common_response.message)
            print("data: ", resp.to_json())
        except Exception as e:
            print("error: ", e)
            raise e

    def test_cancel_all_orders_req(self):
        """
            cancel_all_orders
            Cancel All Orders
            /api/v1/orders
        """

        builder = CancelAllOrdersReqBuilder()
        builder.set_symbol("XBTUSDTM")
        req = builder.build()
        try:
            resp = self.api.cancel_all_orders(req)
            print("code: ", resp.common_response.code)
            print("msg: ", resp.common_response.message)
            print("data: ", resp.to_json())
        except Exception as e:
            print("error: ", e)
            raise e

    def test_get_order_list_req(self):
        """
            get_order_list
            Get Order List
            /api/v1/orders
        """

        builder = GetOrderListReqBuilder()
        builder.set_symbol("XBTUSDTM").set_side(GetOrderListReq.SideEnum.BUY).set_type(
            GetOrderListReq.TypeEnum.LIMIT).set_start_at(1732550400000).set_end_at(1732723200000)
        req = builder.build()
        try:
            resp = self.api.get_order_list(req)
            print("code: ", resp.common_response.code)
            print("msg: ", resp.common_response.message)
            print("data: ", resp.to_json())
        except Exception as e:
            print("error: ", e)
            raise e

    def test_batch_cancel_orders_req(self):
        """
            batch_cancel_orders
            Batch Cancel Orders
            /api/v1/orders/multi-cancel
        """

        o1 = BatchCancelOrdersClientOidsListBuilder().set_client_oid(
            "59872a92-c10d-4027-a7f5-9eebe966eda6").set_symbol("XBTUSDTM").build()
        o2 = BatchCancelOrdersClientOidsListBuilder().set_client_oid(
            "4bd8e0c5-a57a-4627-a23b-3d5f3d87184b").set_symbol("XBTUSDTM").build()

        builder = BatchCancelOrdersReqBuilder()
        builder.set_order_ids_list(['251157545060700160', '251157545257832449']).set_client_oids_list([o1, o2])
        req = builder.build()
        try:
            resp = self.api.batch_cancel_orders(req)
            print("code: ", resp.common_response.code)
            print("msg: ", resp.common_response.message)
            print("data: ", resp.to_json())
        except Exception as e:
            print("error: ", e)
            raise e

    def test_batch_add_orders_req(self):
        """
            batch_add_orders
            Batch Add Orders
            /api/v1/orders/multi
        """

        builder = BatchAddOrdersReqBuilder()

        order1 = BatchAddOrdersItemBuilder().set_client_oid(uuid.uuid4().__str__()).set_side(
            BatchAddOrdersItem.SideEnum.BUY).set_symbol("XBTUSDTM").set_leverage(1).set_type(
            BatchAddOrdersItem.TypeEnum.LIMIT).set_remark("sdk_test").set_margin_mode(
            BatchAddOrdersItem.MarginModeEnum.CROSS).set_price("10000").set_size(1).build()

        order2 = BatchAddOrdersItemBuilder().set_client_oid(uuid.uuid4().__str__()).set_side(
            BatchAddOrdersItem.SideEnum.BUY).set_symbol("XBTUSDTM").set_leverage(1).set_type(
            BatchAddOrdersItem.TypeEnum.LIMIT).set_remark("sdk_test").set_margin_mode(
            BatchAddOrdersItem.MarginModeEnum.CROSS).set_price("10000").set_size(1).build()
        builder.set_items([order1, order2])
        req = builder.build()

        try:
            resp = self.api.batch_add_orders(req)
            print("code: ", resp.common_response.code)
            print("msg: ", resp.common_response.message)
            print("data: ", resp.to_json())
        except Exception as e:
            print("error: ", e)
            raise e

    def test_cancel_order_by_id_req(self):
        """
            cancel_order_by_id
            Cancel Order By OrderId
            /api/v1/orders/{orderId}
        """

        builder = CancelOrderByIdReqBuilder()
        builder.set_order_id("251138988260249600")
        req = builder.build()
        try:
            resp = self.api.cancel_order_by_id(req)
            print("code: ", resp.common_response.code)
            print("msg: ", resp.common_response.message)
            print("data: ", resp.to_json())
        except Exception as e:
            print("error: ", e)
            raise e

    def test_get_order_by_order_id_req(self):
        """
            get_order_by_order_id
            Get Order By OrderId
            /api/v1/orders/{order-id}
        """

        builder = GetOrderByOrderIdReqBuilder()
        builder.set_order_id("251138988260249600")
        req = builder.build()
        try:
            resp = self.api.get_order_by_order_id(req)
            print("code: ", resp.common_response.code)
            print("msg: ", resp.common_response.message)
            print("data: ", resp.to_json())
        except Exception as e:
            print("error: ", e)
            raise e

    def test_add_order_req(self):
        """
            add_order
            Add Order
            /api/v1/orders
        """

        builder = AddOrderReqBuilder()
        builder.set_client_oid(uuid.uuid4().__str__()).set_side(
            AddOrderReq.SideEnum.BUY).set_symbol("XBTUSDTM").set_leverage(1).set_type(
            AddOrderReq.TypeEnum.LIMIT).set_remark("sdk_test").set_margin_mode(
            AddOrderReq.MarginModeEnum.CROSS).set_price("10000").set_size(1)
        req = builder.build()
        try:
            resp = self.api.add_order(req)
            print("code: ", resp.common_response.code)
            print("msg: ", resp.common_response.message)
            print("data: ", resp.to_json())
        except Exception as e:
            print("error: ", e)
            raise e

    def test_add_order_test_req(self):
        """
            add_order_test
            Add Order Test
            /api/v1/orders/test
        """

        builder = AddOrderTestReqBuilder()
        builder.set_client_oid(uuid.uuid4().__str__()).set_side(
            AddOrderTestReq.SideEnum.BUY).set_symbol("XBTUSDTM").set_leverage(1).set_type(
            AddOrderTestReq.TypeEnum.LIMIT).set_remark("sdk_test").set_margin_mode(
            AddOrderTestReq.MarginModeEnum.CROSS).set_price("10000").set_size(1)
        req = builder.build()
        try:
            resp = self.api.add_order_test(req)
            print("code: ", resp.common_response.code)
            print("msg: ", resp.common_response.message)
            print("data: ", resp.to_json())
        except Exception as e:
            print("error: ", e)
            raise e

    def test_get_recent_closed_orders_req(self):
        """
            get_recent_closed_orders
            Get Recent Closed Orders
            /api/v1/recentDoneOrders
        """

        builder = GetRecentClosedOrdersReqBuilder()
        builder.set_symbol("DOGEUSDTM")
        req = builder.build()
        try:
            resp = self.api.get_recent_closed_orders(req)
            print("code: ", resp.common_response.code)
            print("msg: ", resp.common_response.message)
            print("data: ", resp.to_json())
        except Exception as e:
            print("error: ", e)
            raise e

    def test_get_recent_trade_history_req(self):
        """
            get_recent_trade_history
            Get Recent Trade History
            /api/v1/recentFills
        """

        builder = GetRecentTradeHistoryReqBuilder()
        builder.set_symbol("DOGEUSDTM")
        req = builder.build()
        try:
            resp = self.api.get_recent_trade_history(req)
            print("code: ", resp.common_response.code)
            print("msg: ", resp.common_response.message)
            print("data: ", resp.to_json())
        except Exception as e:
            print("error: ", e)
            raise e

    def test_add_tpsl_order_req(self):
        """
            add_tpsl_order
            Add Take Profit And Stop Loss Order
            /api/v1/st-orders
        """

        builder = AddTpslOrderReqBuilder()
        builder.set_client_oid(uuid.uuid4().__str__()).set_side(
            AddTpslOrderReq.SideEnum.BUY).set_symbol("XBTUSDTM").set_leverage(1).set_type(
            AddTpslOrderReq.TypeEnum.LIMIT).set_remark("sdk_test").set_margin_mode(
            AddTpslOrderReq.MarginModeEnum.CROSS).set_price("10000").set_size(1).set_stop_price_type(
            AddTpslOrderReq.StopPriceTypeEnum.TRADE_PRICE).set_trigger_stop_up_price(
            "12000").set_trigger_stop_down_price("8000")
        req = builder.build()
        try:
            resp = self.api.add_tpsl_order(req)
            print("code: ", resp.common_response.code)
            print("msg: ", resp.common_response.message)
            print("data: ", resp.to_json())
        except Exception as e:
            print("error: ", e)
            raise e

    def test_cancel_all_stop_orders_req(self):
        """
            cancel_all_stop_orders
            Cancel All Stop orders
            /api/v1/stopOrders
        """

        builder = CancelAllStopOrdersReqBuilder()
        builder.set_symbol("XBTUSDTM")
        req = builder.build()
        try:
            resp = self.api.cancel_all_stop_orders(req)
            print("code: ", resp.common_response.code)
            print("msg: ", resp.common_response.message)
            print("data: ", resp.to_json())
        except Exception as e:
            print("error: ", e)
            raise e

    def test_get_stop_order_list_req(self):
        """
            get_stop_order_list
            Get Stop Order List
            /api/v1/stopOrders
        """

        builder = GetStopOrderListReqBuilder()
        builder.set_symbol("XBTUSDTM").set_side(GetStopOrderListReq.SideEnum.BUY)
        req = builder.build()
        try:
            resp = self.api.get_stop_order_list(req)
            print("code: ", resp.common_response.code)
            print("msg: ", resp.common_response.message)
            print("data: ", resp.to_json())
        except Exception as e:
            print("error: ", e)
            raise e

File: sdk/golang/tests/e2e/rest/futures_test/futures_orders_test.go
package futures_test

import (
	"context"
	"encoding/json"
	"fmt"
	"github.com/Kucoin/kucoin-universal-sdk/sdk/golang/extension/interceptor"
	"github.com/Kucoin/kucoin-universal-sdk/sdk/golang/pkg/api"
	"github.com/Kucoin/kucoin-universal-sdk/sdk/golang/pkg/common/logger"
	"github.com/Kucoin/kucoin-universal-sdk/sdk/golang/pkg/generate/futures/order"
	"github.com/Kucoin/kucoin-universal-sdk/sdk/golang/pkg/types"
	"github.com/google/uuid"
	"os"
	"testing"
)

var orderApi order.OrderAPI

func init() {
	key := os.Getenv("API_KEY")
	secret := os.Getenv("API_SECRET")
	passphrase := os.Getenv("API_PASSPHRASE")

	defaultLogger := logger.NewDefaultLogger()
	logger.SetLogger(defaultLogger)

	httpOptionBuilder := types.NewTransportOptionBuilder().
		SetKeepAlive(true).
		AddInterceptors(interceptor.NewLoggingInterceptor(false, defaultLogger))

	option := types.NewClientOptionBuilder().
		WithKey(key).
		WithSecret(secret).
		WithPassphrase(passphrase).
		WithSpotEndpoint(types.GlobalApiEndpoint).
		WithFuturesEndpoint(types.GlobalFuturesApiEndpoint).
		WithBrokerEndpoint(types.GlobalBrokerApiEndpoint).
		WithTransportOption(httpOptionBuilder.Build()).
		Build()

	client := api.NewClient(option)

	orderApi = client.RestService().GetFuturesService().GetOrderAPI()
}

func TestFillsGetTradeHistoryReq(t *testing.T) {
	// GetTradeHistory
	// Get Trade History
	// /api/v1/fills

	builder := order.NewGetTradeHistoryReqBuilder()
	builder.SetSide("buy").SetType("market")
	req := builder.Build()

	resp, err := orderApi.GetTradeHistory(req, context.TODO())
	if err != nil {
		panic(err)
	}
	data, err := json.Marshal(resp.ToMap())
	if err != nil {
		panic(err)
	}
	fmt.Println("code:", resp.CommonResponse.Code)
	fmt.Println("message:", resp.CommonResponse.Message)
	fmt.Println("data:", string(data))
}

func TestFillsGetOpenOrderValueReq(t *testing.T) {
	// GetOpenOrderValue
	// Get Open Order Value
	// /api/v1/openOrderStatistics

	builder := order.NewGetOpenOrderValueReqBuilder()
	builder.SetSymbol("DOGEUSDTM")
	req := builder.Build()

	resp, err := orderApi.GetOpenOrderValue(req, context.TODO())
	if err != nil {
		panic(err)
	}
	data, err := json.Marshal(resp.ToMap())
	if err != nil {
		panic(err)
	}
	fmt.Println("code:", resp.CommonResponse.Code)
	fmt.Println("message:", resp.CommonResponse.Message)
	fmt.Println("data:", string(data))
}

func TestOrderGetOrderByClientOidReq(t *testing.T) {
	// GetOrderByClientOid
	// Get Order By ClientOid
	// /api/v1/orders/byClientOid

	builder := order.NewGetOrderByClientOidReqBuilder()
	builder.SetClientOid("df2e52bb-1d11-44d9-91a9-26899f3f46cb")
	req := builder.Build()

	resp, err := orderApi.GetOrderByClientOid(req, context.TODO())
	if err != nil {
		panic(err)
	}
	data, err := json.Marshal(resp.ToMap())
	if err != nil {
		panic(err)
	}
	fmt.Println("code:", resp.CommonResponse.Code)
	fmt.Println("message:", resp.CommonResponse.Message)
	fmt.Println("data:", string(data))
}

func TestOrderCancelOrderByClientOidReq(t *testing.T) {
	// CancelOrderByClientOid
	// Cancel Order By ClientOid
	// /api/v1/orders/client-order/{clientOid}

	builder := order.NewCancelOrderByClientOidReqBuilder()
	builder.SetSymbol("XBTUSDTM").SetClientOid("e59071a0-382e-45f0-b934-6c0cea16908b")
	req := builder.Build()

	resp, err := orderApi.CancelOrderByClientOid(req, context.TODO())
	if err != nil {
		panic(err)
	}
	data, err := json.Marshal(resp.ToMap())
	if err != nil {
		panic(err)
	}
	fmt.Println("code:", resp.CommonResponse.Code)
	fmt.Println("message:", resp.CommonResponse.Message)
	fmt.Println("data:", string(data))
}

func TestOrderCancelAllOrdersV1Req(t *testing.T) {
	// CancelAllOrders
	// Cancel All Orders
	// /api/v1/orders

	builder := order.NewCancelAllOrdersV1ReqBuilder()
	builder.SetSymbol("XBTUSDTM")
	req := builder.Build()

	resp, err := orderApi.CancelAllOrdersV1(req, context.TODO())
	if err != nil {
		panic(err)
	}
	data, err := json.Marshal(resp.ToMap())
	if err != nil {
		panic(err)
	}
	fmt.Println("code:", resp.CommonResponse.Code)
	fmt.Println("message:", resp.CommonResponse.Message)
	fmt.Println("data:", string(data))
}
func TestOrderGetOrderListReq(t *testing.T) {
	// GetOrderList
	// Get Order List
	// /api/v1/orders

	builder := order.NewGetOrderListReqBuilder()
	builder.SetStatus("active").SetSymbol("XBTUSDTM").SetSide("buy")
	req := builder.Build()

	resp, err := orderApi.GetOrderList(req, context.TODO())
	if err != nil {
		panic(err)
	}
	data, err := json.Marshal(resp.ToMap())
	if err != nil {
		panic(err)
	}
	fmt.Println("code:", resp.CommonResponse.Code)
	fmt.Println("message:", resp.CommonResponse.Message)
	fmt.Println("data:", string(data))
}

func TestOrderBatchCancelOrdersReq(t *testing.T) {
	// BatchCancelOrders
	// Batch Cancel Orders
	// /api/v1/orders/multi-cancel

	builder := order.NewBatchCancelOrdersReqBuilder()
	builder.SetOrderIdsList([]string{"250442580993564673", "250442581069062144"}).SetClientOidsList([]order.BatchCancelOrdersClientOidsList{
		*order.NewBatchCancelOrdersClientOidsList("XBTUSDTM", "4d7571f4-8a26-43a0-9fae-a49161682d4a"),
		*order.NewBatchCancelOrdersClientOidsList("XBTUSDTM", "8ae42301-86d5-4df5-8c43-86561d7e8b13"),
	})
	req := builder.Build()

	resp, err := orderApi.BatchCancelOrders(req, context.TODO())
	if err != nil {
		panic(err)
	}
	data, err := json.Marshal(resp.ToMap())
	if err != nil {
		panic(err)
	}
	fmt.Println("code:", resp.CommonResponse.Code)
	fmt.Println("message:", resp.CommonResponse.Message)
	fmt.Println("data:", string(data))
}
func TestOrderBatchAddOrdersReq(t *testing.T) {
	// BatchAddOrders
	// Batch Add Orders
	// /api/v1/orders/multi

	order1 := order.NewBatchAddOrdersItemBuilder().SetClientOid(uuid.NewString()).SetSide("buy").SetSymbol("XBTUSDTM").
		SetLeverage(3.0).SetType("limit").SetRemark("order_test").SetMarginMode("CROSS").
		SetPrice("10000").SetSize(1).Build()
	order2 := order.NewBatchAddOrdersItemBuilder().SetClientOid(uuid.NewString()).SetSide("buy").SetSymbol("XBTUSDTM").
		SetLeverage(3.0).SetType("limit").SetRemark("order_test").SetMarginMode("CROSS").
		SetPrice("10000").SetSize(1).Build()

	builder := order.NewBatchAddOrdersReqBuilder()
	builder.SetItems([]order.BatchAddOrdersItem{*order1, *order2})
	req := builder.Build()

	resp, err := orderApi.BatchAddOrders(req, context.TODO())
	if err != nil {
		panic(err)
	}
	data, err := json.Marshal(resp.ToMap())
	if err != nil {
		panic(err)
	}
	fmt.Println("code:", resp.CommonResponse.Code)
	fmt.Println("message:", resp.CommonResponse.Message)
	fmt.Println("data:", string(data))
}

func TestOrderCancelOrderByIdReq(t *testing.T) {
	// CancelOrderById
	// Cancel Order By OrderId
	// /api/v1/orders/{orderId}

	builder := order.NewCancelOrderByIdReqBuilder()
	builder.SetOrderId("250442004150878209")
	req := builder.Build()

	resp, err := orderApi.CancelOrderById(req, context.TODO())
	if err != nil {
		panic(err)
	}
	data, err := json.Marshal(resp.ToMap())
	if err != nil {
		panic(err)
	}
	fmt.Println("code:", resp.CommonResponse.Code)
	fmt.Println("message:", resp.CommonResponse.Message)
	fmt.Println("data:", string(data))
}
func TestOrderGetOrderByOrderIdReq(t *testing.T) {
	// GetOrderByOrderId
	// Get Order By OrderId
	// /api/v1/orders/{order-id}

	builder := order.NewGetOrderByOrderIdReqBuilder()
	builder.SetOrderId("250441187108892672")
	req := builder.Build()

	resp, err := orderApi.GetOrderByOrderId(req, context.TODO())
	if err != nil {
		panic(err)
	}
	data, err := json.Marshal(resp.ToMap())
	if err != nil {
		panic(err)
	}
	fmt.Println("code:", resp.CommonResponse.Code)
	fmt.Println("message:", resp.CommonResponse.Message)
	fmt.Println("data:", string(data))
}
func TestOrderAddOrderReq(t *testing.T) {
	// AddOrder
	// Add Order
	// /api/v1/orders

	builder := order.NewAddOrderReqBuilder()
	builder.SetClientOid(uuid.NewString()).SetSide("buy").SetSymbol("XBTUSDTM").
		SetLeverage(3.0).SetType("limit").SetRemark("order_test").SetMarginMode("CROSS").
		SetPrice("10000").SetSize(1)
	req := builder.Build()

	resp, err := orderApi.AddOrder(req, context.TODO())
	if err != nil {
		panic(err)
	}
	data, err := json.Marshal(resp.ToMap())
	if err != nil {
		panic(err)
	}
	fmt.Println("code:", resp.CommonResponse.Code)
	fmt.Println("message:", resp.CommonResponse.Message)
	fmt.Println("data:", string(data))
}
func TestOrderAddOrderTestReq(t *testing.T) {
	// AddOrderTest
	// Add Order Test
	// /api/v1/orders/test

	builder := order.NewAddOrderTestReqBuilder()
	builder.SetClientOid(uuid.NewString()).SetSide("buy").SetSymbol("XBTUSDTM").
		SetLeverage(3.0).SetType("limit").SetRemark("order_test").SetMarginMode("ISOLATED").
		SetPrice("10000").SetSize(1)
	req := builder.Build()

	resp, err := orderApi.AddOrderTest(req, context.TODO())
	if err != nil {
		panic(err)
	}
	data, err := json.Marshal(resp.ToMap())
	if err != nil {
		panic(err)
	}
	fmt.Println("code:", resp.CommonResponse.Code)
	fmt.Println("message:", resp.CommonResponse.Message)
	fmt.Println("data:", string(data))
}

func TestOrderGetRecentClosedOrdersReq(t *testing.T) {
	// GetRecentClosedOrders
	// Get Recent Closed Orders
	// /api/v1/recentDoneOrders

	builder := order.NewGetRecentClosedOrdersReqBuilder()
	builder.SetSymbol("XBTUSDTM")
	req := builder.Build()

	resp, err := orderApi.GetRecentClosedOrders(req, context.TODO())
	if err != nil {
		panic(err)
	}
	data, err := json.Marshal(resp.ToMap())
	if err != nil {
		panic(err)
	}
	fmt.Println("code:", resp.CommonResponse.Code)
	fmt.Println("message:", resp.CommonResponse.Message)
	fmt.Println("data:", string(data))
}

func TestFillsGetRecentTradeHistoryReq(t *testing.T) {
	// GetRecentTradeHistory
	// Get Recent Trade History
	// /api/v1/recentFills

	builder := order.NewGetRecentTradeHistoryReqBuilder()
	builder.SetSymbol("DOGEUSDTM")
	req := builder.Build()

	resp, err := orderApi.GetRecentTradeHistory(req, context.TODO())
	if err != nil {
		panic(err)
	}
	data, err := json.Marshal(resp.ToMap())
	if err != nil {
		panic(err)
	}
	fmt.Println("code:", resp.CommonResponse.Code)
	fmt.Println("message:", resp.CommonResponse.Message)
	fmt.Println("data:", string(data))
}

func TestOrderAddTPSLOrderReq(t *testing.T) {
	// AddTPSLOrder
	// Add Take Profit And Stop Loss Order
	// /api/v1/st-orders

	builder := order.NewAddTPSLOrderReqBuilder()
	builder.SetClientOid(uuid.NewString()).SetSide("buy").SetSymbol("XBTUSDTM").SetMarginMode("CROSS").
		SetLeverage(1.0).SetType("limit").SetRemark("test_order").SetStopPriceType("TP").
		SetPrice("10000").SetSize(1).SetTriggerStopUpPrice("8000").SetTriggerStopDownPrice("12000")
	req := builder.Build()

	resp, err := orderApi.AddTPSLOrder(req, context.TODO())
	if err != nil {
		panic(err)
	}
	data, err := json.Marshal(resp.ToMap())
	if err != nil {
		panic(err)
	}
	fmt.Println("code:", resp.CommonResponse.Code)
	fmt.Println("message:", resp.CommonResponse.Message)
	fmt.Println("data:", string(data))
}

func TestOrderCancelAllStopOrdersReq(t *testing.T) {
	// CancelAllStopOrders
	// Cancel All Stop orders
	// /api/v1/stopOrders

	builder := order.NewCancelAllStopOrdersReqBuilder()
	builder.SetSymbol("XBTUSDTM")
	req := builder.Build()

	resp, err := orderApi.CancelAllStopOrders(req, context.TODO())
	if err != nil {
		panic(err)
	}
	data, err := json.Marshal(resp.ToMap())
	if err != nil {
		panic(err)
	}
	fmt.Println("code:", resp.CommonResponse.Code)
	fmt.Println("message:", resp.CommonResponse.Message)
	fmt.Println("data:", string(data))
}

func TestOrderGetStopOrderListReq(t *testing.T) {
	// GetStopOrderList
	// Get Stop Order List
	// /api/v1/stopOrders

	builder := order.NewGetStopOrderListReqBuilder()
	builder.SetSymbol("XBTUSDTM")
	req := builder.Build()

	resp, err := orderApi.GetStopOrderList(req, context.TODO())
	if err != nil {
		panic(err)
	}
	data, err := json.Marshal(resp.ToMap())
	if err != nil {
		panic(err)
	}
	fmt.Println("code:", resp.CommonResponse.Code)
	fmt.Println("message:", resp.CommonResponse.Message)
	fmt.Println("data:", string(data))
}

func TestOrderCancelAllOrdersV3Req(t *testing.T) {
	// CancelAllOrders
	// Cancel All Orders
	// /api/v1/orders

	builder := order.NewCancelAllOrdersV3ReqBuilder()
	builder.SetSymbol("XBTUSDTM")
	req := builder.Build()

	resp, err := orderApi.CancelAllOrdersV3(req, context.TODO())
	if err != nil {
		panic(err)
	}
	data, err := json.Marshal(resp.ToMap())
	if err != nil {
		panic(err)
	}
	fmt.Println("code:", resp.CommonResponse.Code)
	fmt.Println("message:", resp.CommonResponse.Message)
	fmt.Println("data:", string(data))
}
